API/NoDocs/Microsoft.Extensions.Http.Resilience/Endpoint.cs

```diff
@@ -0,0 +1,13 @@
+// Assembly 'Microsoft.Extensions.Http.Resilience'
+
+using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace Microsoft.Extensions.Http.Resilience;
+
+public class Endpoint
```

Comment:
@martintmk Thoughts?

Reply:
We could consider changing the name. For example:

- `HttpEndpoint`
- `UriEndpoint`
- `EndpointAddress`

But it should be paired with renaming all relevant APIs

- Endpoint
- EndpointGroup
- WeightedEndpoint
- WeightedEnpointGroup

In that case the type names might get too "wordy". 

Isn't the namespace discriminator `Microsoft.Extensions.Http.Resilience` enough?

Reply:
> Isn't the namespace discriminator Microsoft.Extensions.Http.Resilience enough?

The problem is when multiple namespaces are imported.

So someone has:
```C#
using System.Net;
using Microsoft.Extensions.Http.Resilience;
```
and now you have two EndPoint types (although, even more confusing, the casing is different, as the System.Net type is EndPoint and this one is Endpoint).

> In that case the type names might get too "wordy".

In general, we're ok with wordier where it's warranted. I think it's warranted here.

Reply:
> In general, we're ok with wordier where it's warranted. I think it's warranted here.

That's fair, any tips for rename? Maybe just `UriEndpoint` since it only exposes a single `Uri` property?

Reply:
Done.